docs/docs/dynamo-store/architecture.md

```diff
@@ -0,0 +1,117 @@
+---
+sidebar_position: 1
+---
+
+# DynamoStore Architecture
+
+When it comes to building an event store on top of a serverless NoSQL database
+it is imperative to apply mechanical sympathy. We've seen implementations of
+DynamoDB event stores using a single-event-per-row strategy. This would seem the
+obvious choice so let's start by explaining why it's not the greatest idea.
+
+## DynamoDB cost model
+
+A great article to understand DynamoDB's pricing model was written by Zach
+Charles
+[here](https://zaccharles.medium.com/calculating-a-dynamodb-items-size-and-consumed-capacity-d1728942eb7c).
+
+The gist of it is that you pay per 4KB for reads, and 1KB for writes, and
+transactional writes will cost you double. As such DynamoStore is designed to
+utilise GetItem and PutItem APIs where possible and only does a table scan and
+transactional write when absolutely necessary.
+
+With these constraints enumerated it should be clear why an event-per-row
+schema is not the greatest idea
+
+1. Reading a stream necessitates a scan in which you pay 0.5RUs at minimum for
+   each event in the stream
+2. Writing will always be transactional so you'll pay double
+
+## The architecture
+
+At heart DynamoDB is a document store and an event sourced system is composed
+of multiple streams. It therefore stands to reason that we should model the
+storage such that the stream is the document. This works great until we run into
```

Comment:
"the stream is the document" -> "that the events of the stream are co-located in a document"
I'd also consider adding a sentence that says that in a doc store you want to chop things into units that you want/need to write and/or read (and that for event sourcing, there is no case where you need to load/update/insert one event independent of all others).

Reply:
Should I reword and use `Item` instead of document, and say DDB is a KV store at heart?

Reply:
I'd say its more than just KV, and the point is to leverage all that implies
But I think that is a good way to put it - "As with any KV store, point reads of a single item at a time are its sweet spot"
I dont think there'a any harm in invoking the term document once early on
then, KV store
and from then on switch to Partition of colocated Items sorted by the SortKey
And maybe work in the fact that DDB presents two addressable things
- Partition, which can be scanned or queried
- Item, which can be transacted against
Anything else: multiple writes, scattered reads, transactional updates is going through an extra layer, which means jitter, latency and/or charges.

I guess this also leads onto explaining that DDB streams is a scalable tree that represents all writes/deletes to Items over the last 24h. The only guarantee you get is that updates to any individual items will come out the other end in the order they were applied. 

When you think about it/put it that way, one wouldn't have any expectation of any relationship between the proceeds of updates to multiple items 
in a transactional batch update as they pop out the other end